backend/layers/business/business.py

```diff
@@ -943,6 +943,14 @@ def get_unpublished_dataset_versions(self, dataset_id: DatasetId) -> List[Datase
             )
         )
 
+    def delete_dataset_versions(self, dataset_versions: List[DatasetVersion]) -> None:
+        """
+        Deletes a list of dataset versions and associated dataset artifact rows from the database, as well
+        as kicking off deletion of their corresponding assets from S3
+        """
+        self.database_provider.delete_dataset_versions(dataset_versions)
```

Comment:
We should delete the s3 assets first. It's easier to find stray database rows than it is to find stray s3 assets. enforcing the asset deletion before the database row ensures we have a pointer to the assets incase assets deletion fails for some reason.